clients/src/main/java/org/apache/kafka/common/metrics/stats/Avg.java

```diff
@@ -26,23 +26,32 @@
 public class Avg extends SampledStat {
 
     public Avg() {
-        super(0.0);
+        super(Double.NaN);
     }
 
     @Override
     protected void update(Sample sample, MetricConfig config, double value, long now) {
-        sample.value += value;
+        if (Double.isNaN(sample.value))
+            sample.value = value;
+        else
+            sample.value += value;
     }
 
     @Override
     public double combine(List<Sample> samples, MetricConfig config, long now) {
-        double total = 0.0;
+        double total = Double.NaN;
         long count = 0;
         for (Sample s : samples) {
-            total += s.value;
+            if (Double.isNaN(s.value))
+                continue;
+            if (Double.isNaN(total))
+                total = s.value;
+            else
+                total += s.value;
+
             count += s.eventCount;
         }
-        return count == 0 ? 0 : total / count;
+        return count == 0 ? total : total / count;
```

Comment:
Hi @stanislavkozlovski ,

Thanks for this PR. If I'm reading the code of SampledStat right, I think you could achieve your objective simply by replacing this line with:
```
return count == 0 ? Double.NaN : total / count;
```

And then, you wouldn't need any of the other changes in this class. Does that seem right to you?

If so, I think it would apply to the rest of the changed stats as well.

Thanks,
-John

Reply:
Yes, that would be the same functionality. Do you think that makes the code clearer?

Reply:
Yeah, personally, I think so.

As long as we can report NaN externally for un-initialized metrics, it really doesn't matter how we initialize each sample internally. IMHO, each metric should use an initial sample value that makes its internal math simple, since those samples aren't directly visible externally. (unless I've misunderstood the system)

Reply:
That makes a lot of sense. Can you take another look, @vvcephei ?